recipes/ndb.rb

```diff
@@ -0,0 +1,95 @@
+include_recipe "cloud::default"
+
+directory node['cloud']['ndb-agent']['base_dir'] do
+    owner "root"
+    group "root"
+    mode "700"
+    recursive true
+end
+
+link node['cloud']['ndb-agent']['home'] do
+    owner 'root'
+    group 'root'
+    to node['cloud']['ndb-agent']['base_dir']
+end
+
+directory "#{node['cloud']['ndb-agent']['bin']}" do
+    owner "root"
+    group "root"
+    mode "700"
+end
+
+directory "#{node['cloud']['ndb-agent']['config']}" do
+    owner "root"
+    group "root"
+    mode "700"
+end
+
+filename = "ndb-agent.tgz"
+source = "#{node['install']['enterprise']['download_url']}/ndb-agent/#{node['cloud']['ndb-agent']['version']}/#{filename}"
+remote_file "#{Chef::Config['file_cache_path']}/#{filename}" do
+    user 'root'
+    group 'root'
+    source source
+    headers get_ee_basic_auth_header()
+    sensitive true
+    mode 0550
+    action :create_if_missing
+end
+
+bash "Setup binaries" do
+    user 'root'
+    group 'root'
+    code <<-EOH
+        set -e
+        tar xzf #{Chef::Config['file_cache_path']}/#{filename} -C #{Chef::Config['file_cache_path']}
+        cp #{Chef::Config['file_cache_path']}/ndb-agent/ndb-agent #{node['cloud']['ndb-agent']['bin']}
+        chmod 750 #{node['cloud']['ndb-agent']['bin']}/ndb-agent
+
+        cp -r #{Chef::Config['file_cache_path']}/ndb-agent/contrib #{node['cloud']['ndb-agent']['bin']}
+        chmod -R 750 #{node['cloud']['ndb-agent']['bin']}/contrib
+    EOH
```

Comment:
I guess it is better to add a condition only_if or not_if here

Reply:
I added a guard